tests/test_catalog.py

```diff
@@ -85,6 +86,19 @@ def test_create_and_read(self) -> None:
 
             self.assertEqual(len(list(items)), 8)
 
+    def test_from_dict_preserves_dict(self) -> None:
+        catalog_dict = TestCases.test_case_1().to_dict()
+        param_dict = deepcopy(catalog_dict)
+
+        # test that the parameter is preserved
+        _ = Catalog.from_dict(param_dict)
+        self.assertEqual(param_dict, catalog_dict)
```

Comment:
May want to use `assertIs` here (and `assertIsNot` below) in case we update equality per #380. Same for a few other assertions below.

Reply:
@duckontheweb these are dicts, not PySTAC objects, so equals makes sense here right?

Reply:
Shoot, totally missed that. Yes, carry on!